### ISSUE TITLE
nvim-qt opens terminal window on start

### ISSUE THREAD

**Issue by equalsraf** [equalsraf/neovim-qt#93]
From https://github.com/equalsraf/neovim-qt/issues/59#issuecomment-179853176, opening the nvim-qt binary also opens a terminal window. Apparently this is standard behaviour in Mac OS ([here](http://stackoverflow.com/questions/30713057/do-executable-files-always-open-a-terminal-window-on-macos)).

I'm not sure if bundling the manifest with the binary is enough to fix this issue, but CMake does have a module to do it https://cmake.org/cmake/help/v3.0/prop_tgt/MACOSX_BUNDLE_INFO_PLIST.html

cc @rkitover

**Comment by james2doyle** [comment]
Here is what happens when I start nvim-qt from scratch.

![terminal window opens](http://cl.ly/2T0g2j3B2i2C/Screen%20Recording%202016-02-04%20at%2006.53%20AM.gif)

You can see there is some sort of debug log running in stdout

**Comment by rkitover** [comment]
@equalsraf I think we are going to need to make a simple .app bundle. Perhaps not a fully independent one yet, but at least enough to hide the terminal window. Want me to take a crack at it?

**Comment by equalsraf** [comment]
@rkitover If you get the chance please do :D

**Comment by dvcrn** [comment]
For future readers, it is easy to get around this without a .app bundle by using `open`. Here's a snippet from my dotfiles:

```
gnvim() {
  ABSOLUTEPATH=`realpath $1`
  open -n -a ~/src/neovim-qt/build/bin/nvim-qt --args -- $ABSOLUTEPATH
}
```

```
gnvim foo.sh
```

will open neovim-qt without the shell as if it were a normal mac app

**Comment by equalsraf** [comment]
Started something in https://github.com/equalsraf/neovim-qt/pull/143 and the shell is gone. But it fails to find the nvim executable (/usr/local/ not in PATH?)

**Comment by equalsraf** [comment]
Ok I've cleaned up #143, the one remaining issue is that the bundle does not display the correct icon. Can anyone test it and confirm?

PS1: I have not done anything yet for the Mac file open events so some things like , `open -a nvim-qt FILENAME` don't work yet, but I would its best to deal with that in a separate PR.

PS2: it seems that macdeployqt can generate a bundle that also include the Qt libraries, I was able to do this and even generate a dmg but it failed to run, apparently conflicted with the Qt installed under /usr/local. There is a commented entry in src/gui/CMakeLists.txt for that if anyone is interested.
